molecule_vagrant/modules/vagrant.py

```diff
@@ -185,6 +195,8 @@
     c.vm.box = "{{ instance.box }}"
     {{ 'c.vm.box_version = "{}"'.format(instance.box_version) if instance.box_version }}
     {{ 'c.vm.box_url = "{}"'.format(instance.box_url) if instance.box_url }}
+    {{ 'c.vm.box_download_checksum = "{}"'.format(instance.box_download_checksum) if instance.box_download_checksum }}
+    {{ 'c.vm.box_download_checksum_type = "{}"'.format(instance.box_download_checksum_type) if instance.box_download_checksum_type }}
```

Comment:
Did you check that if someone has specified a checksum type and but forget to specify a checksum, things are still working ?
The other solution would be to use a if block testing if ``box_download_checksum`` is set and in this case, set
``c.vm.box_download_checksum`` and ``c.vm.box_download_checksum_type`` ? With the if block, one has still to make sure the checksum type has been specified, but this can be easily handled (using a ``required_if`` or using a default checksum type).

Reply:
The errors generated are:

```
"msg": "Failed to validate generated Vagrantfile: b'There are errors in the configuration of this machine. Please fix\\nthe following errors and try again:\\n\\nvm:\\n* Checksum type specified but \"box_download_checksum\" is blank\\n\\n'"
```
and
```
"msg": "Failed to validate generated Vagrantfile: b'There are errors in the configuration of this machine. Please fix\\nthe following errors and try again:\\n\\nvm:\\n* Checksum specified but must also specify \"box_download_checksum_type\"\\n\\n'"
```

I believe those are valid and pretty straight forward. Would a requirement improve this?

Reply:
yeah, there are valid errors and I think that it would actually be better to fail earlier if one of the arguments is missing. I suggested ``required_if`` but with more coffee, I guess that ``required_together`` is a better option. You'll have to double check but this may solve these two errors and avoid to use a if block here. A if block here may make things a little bit easier to understand/read.

Reply:
```
failed: [localhost] (item=centos-stream) => {"ansible_loop_var": "item", "changed": false, "item": {"box": "centos-stream/20201019", "box_download_checksum": "96d388ff2170430f032149323ffbba20309b8e0da924e1ad6ec5f024936c2648", "box_url": "https://cloud.centos.org/centos/8-stream/x86_64/images/CentOS-Stream-Vagrant-8-20201019.1.x86_64.vagrant-virtualbox.box", "config_options": {"vm.boot_timeout": 600}, "instance_raw_config_args": ["vbguest.installer_options = { allow_kernel_upgrade: true }"], "memory": 1024, "name": "centos-stream"}, "msg": "parameters are required together: platform_box_download_checksum, platform_box_download_checksum_type"}
```